services/api/src/models/group.ts

```diff
@@ -156,250 +216,297 @@ export const Group = (clients: {
 
-    try {
-      group = await getGroupCacheByName(name)
-    } catch(err) {
-      logger.warn(`Error reading redis, falling back to direct lookup: ${err.message}`);
+    if (!keycloakGroup) {
+      throw new GroupNotFoundError(`Group not found: ${id}`);
     }
 
-    if (!group) {
-      const keycloakGroups = await keycloakAdminClient.groups.find({
-        search: name
-      });
+    return createGroupFromKeycloak(keycloakGroup);
+  };
 
-      if (R.isEmpty(keycloakGroups)) {
-        throw new GroupNotFoundError(`Group not found: ${name}`);
-      }
+  /**
+   * Loads a "full" group from keycloak.
+   * @deprecated Use `loadSparseGroupById` instead.
+   */
+  const loadGroupById = async (id: string): Promise<Group> => {
+    let fullGroup: Group | null = null;
 
-      // Use mutable operations to avoid running out of heap memory
-      const flattenGroups = (groups, group) => {
-        groups.push(R.omit(['subGroups'], group));
-        const flatSubGroups = group.subGroups.reduce(flattenGroups, []);
-        return groups.concat(flatSubGroups);
-      };
+    try {
+      fullGroup = await getGroupCacheById(id);
+    } catch (err: unknown) {
+      if (err instanceof RedisCacheLoadError) {
+        logger.info(err.message);
+      } else {
+        throw err;
+      }
+    }
 
-      group = R.pipe(
-        R.reduce(flattenGroups, []),
-        R.filter(R.propEq('name', name))
-      )(keycloakGroups);
+    if (!fullGroup) {
+      const keycloakGroup = (await keycloakAdminClient.groups.findOne({
+        id,
+      })) as KeycloakLagoonGroup;
 
-      if (R.isEmpty(group)) {
-        throw new GroupNotFoundError(`Group not found: ${name}`);
+      if (!keycloakGroup) {
+        throw new GroupNotFoundError(`Group not found: ${id}`);
       }
 
-      const keycloakGroup = await keycloakAdminClient.groups.findOne({
-        id: group[0].id,
-        briefRepresentation: false,
-      });
-      const groups = await transformKeycloakGroups([keycloakGroup]);
-      group = groups[0];
+      const groupWithSubgroups = await loadKeycloakChildren(keycloakGroup);
+      const groups = await transformKeycloakGroups([groupWithSubgroups]);
+      fullGroup = groups[0];
+
       try {
-        await saveGroupCache(group)
-      } catch (err) {
-        logger.info (`Error saving redis: ${err.message}`)
+        await saveGroupCache(fullGroup);
+      } catch (err: unknown) {
+        if (err instanceof RedisCacheSaveError) {
+          logger.info(err.message);
+        } else {
+          throw err;
+        }
       }
     }
 
-    // @ts-ignore
-    return group;
+    return fullGroup;
   };
 
-  const loadGroupByIdOrName = async (
-    groupInput: GroupInput
-  ): Promise<Group> => {
-    if (R.prop('id', groupInput)) {
-      return loadGroupById(R.prop('id', groupInput));
+  // Load a single, lightweight group (no subgroups/members).
+  const loadSparseGroupByName = async (name: string): Promise<SparseGroup> => {
```

Comment:
Since this and the `ById` version aren't using the redis cache, you're pretty confident there won't be any performance issues with recalling information directly from keycloak in production?

Reply:
I'm ok with merging without redis on these, we can always add redis if we encounter issues.